pnc/src/main/java/org/jboss/pnc/bacon/pnc/common/ClientCreator.java

```diff
@@ -0,0 +1,72 @@
+package org.jboss.pnc.bacon.pnc.common;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.jboss.pnc.bacon.pnc.client.PncClientHelper;
+import org.jboss.pnc.client.ClientBase;
+import org.jboss.pnc.client.Configuration;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+
+/**
+ * Helper class to create the PNC CLI client to use in the CLI objects. There are 2 variants, the unauthenticated one and the
+ * authenticated one
+ *
+ * @param <T> : PNC client to initialize
+ */
+@Slf4j
+public class ClientCreator<T extends ClientBase> {
+
+    private Class<T> tClass;
+
+    private T client;
+    private T clientAuthenticated;
+
+    /**
+     * While Java Generics is nice, there's no way of getting the class object of a generic type. As such, we need to find a way
+     * to capture the Generic class, hence the constructor to capture it. It's sad I know
+     */
+    public ClientCreator(Class<T> tClass) {
```

Comment:
If you don't want to use reflection, you can accept `Supplier<T>` for the constructor method and then create ClientCreator as:
```
new ClientCreator<>(ProjectClient::new);
```

Not sure if that is better, but it's an option.

Reply:
I'll try it. Getting rid of reflection is better in case we want to compile it with Graal. Thanks for the suggestion buddy! :)